aten/src/ATen/TensorGeometry.h

```diff
@@ -113,6 +113,27 @@ struct TORCH_API TensorGeometry {
     return r;
   }
 
+  std::vector<c10::SymInt>& mutable_sizes() {
```

Comment:
Does this ever get mutated? It's called `mutable_sizes` and isn't const-ified, but I only see it used in `before` and `after` of compiled_autograd. The place where you do iterate, you access `sizes_` directly.

Do you want to constify it?

Reply:
Yes, it gets mutated in order to support dynamic shapes in apply_with_saved().  We will temporarily swap in SymInts in `before()`, then restore the old values in `after()`.

The mutation happens through a visitor pattern that walks the Node structure and modifies everything in-place.  Creating copies of nodes would break the graph data structures, be slower, and a lot more code.

Reply:
Yeah sgtm, wasn't sure it was getting mutated.